File: read_pdf.py
import pypdf

reader = pypdf.PdfReader('final_PROJECT_grading-factor.pdf')
text = ""
for page in reader.pages:
    text += page.extract_text() + "\n"

with open('grading_factor.txt', 'w', encoding='utf-8') as f:
    f.write(text)

print("PDF text extracted to grading_factor.txt")

File: read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

let dataBuffer = fs.readFileSync('final_PROJECT_grading-factor.pdf');

pdf(dataBuffer).then(function(data) {
    fs.writeFileSync('grading_factor.txt', data.text);
    console.log('PDF text extracted to grading_factor.txt');
});
